substrate-node/pallets/pallet-tft-price/src/lib.rs

```diff
@@ -271,6 +333,25 @@ impl<T: Config> Module<T> {
         Some(price.integer as u32 * 1000 + (price.fraction / 10_u64.pow(exp)) as u32)
     }
 
+    /// Parse the lowest price from the given JSON string using `serde_json`.
+    ///
+    /// Returns `None` when parsing failed or `Some(price in mUSD)` when parsing is successful.
+    pub fn parse_lowest_price_from_request(price_str: &str) -> Option<u32> {
+        let data: Value = serde_json::from_str(price_str).ok()?;
+        let records_array = data.get("_embedded")?.get("records")?;
+
+        let mut prices = Vec::new();
+        for item in records_array.as_array().unwrap() {
```

Comment:
`unwrap` is kinda dangerous, it should at all times be avoided.

I think you can write this statement using `.map(..)` see https://doc.rust-lang.org/std/iter/struct.Map.html